@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -672,6 +681,23 @@ function placePointLabelAtAnchor(
         // It might be changed if we would like to render text without icon (at border, etc.).
         return persistent ? PlacementResult.Rejected : PlacementResult.Invisible;
     }
+
+    if (measureText) {
+        // Bigger label margin is applied only for the first time it is placed and its visible
+        // with its normal bounds. This margin compensates the changes of label bounds due to
+        // distance scaling and it is designed to cover size changes due to camera panning.
+        // This ways label "reserves" enough space so it may "grow" without affecting other
+        // labels, no collisions occurs and thus limits labels flickering on pan.
+        tmpBox.getSize(tmpSize);
+        tmpSize.multiplyScalar(scale * (1 + newPointLabelTextMarginPercent));
+        tmp2DBox.set(
+            tmpCenter.x - tmpSize.x / 2,
+            tmpCenter.y - tmpSize.y / 2,
+            tmpSize.x,
+            tmpSize.y
+        );
```

Comment:
Here you're recomputing the bbox for new labels again. I guess you're doing this because you want to use the bbox without extra margins to check the label visibility (line 672), is that right? 
I'd rather compute the enlarged bbox for new labels at the top once, and apply it even for visibility check.
What i mean is, just apply the proper scale in line 664 (scale for persistent labels, scale * (1 + newPointLabelTextMarginPercent) for new labels).

Reply:
Believe me I have tried this one before, bigger margin for visibility makes new labels visible even if they are not, in the next frame they become invisible again - labels flickers, but behind the scene... It also affects a lot of unit test which are then failing - because of this change in logic.
It has much more sense to apply it after visibility checking, because we do not try to display unnecessary labels, it gives more consistency between first and next frames and the last but not least unit tests logic is preserved.

Reply:
ok.